pulsar-metadata/src/main/java/org/apache/pulsar/metadata/cache/impl/MetadataCacheImpl.java

```diff
@@ -108,6 +108,9 @@ private CompletableFuture<Optional<CacheGetResult<T>>> readValueFromStore(String
 
                     try {
                         GetResult res = optRes.get();
+                        if (res.getValue().length == 0) {
+                            return FutureUtils.value(Optional.of(new CacheGetResult<>(null, res.getStat())));
+                        }
```

Comment:
There would be less code duplication if this change would be handled in a way where `obj` is set to null when `res.getValue().length == 0`

Reply:
for example `T obj = res.getValue().length > 0 ? serde.deserialize(path, res.getValue(), res.getStat()) : null;`

Reply:
BookieServiceInfo handled the case of `res.getValue.length==0` in deserialize
https://github.com/apache/pulsar/blob/1b43b9de72ccf57b56375870b43d551def59d8a3/pulsar-metadata/src/main/java/org/apache/pulsar/metadata/bookkeeper/BookieServiceInfoSerde.java#L72-L74

That's why the test TestAutoRecoveryAlongWithBookieServers#testAutoRecoveryAlongWithBookieServers failed.

Maybe we need to deserialize first and add the `res.getValue.length==0` check in the exception section?

Reply:
@hangc0276 Thank you! I realized I was wrong to do this fix here because the byte[0] may be a magic value for the SerDe implementation. It should be handled in the SerDe classes. I will close this PR.